src/joystick/SDL_gamepad.c

```diff
@@ -2574,7 +2574,12 @@ bool SDL_IsGamepad(SDL_JoystickID instance_id)
     SDL_LockJoysticks();
     {
         const void *value;
-        if (SDL_FindInHashTable(s_gamepadInstanceIDs, (void *)(uintptr_t)instance_id, &value)) {
+        SDL_JoystickType js_type = SDL_GetJoystickTypeForID(instance_id);
```

Comment:
You probably want this logic in the else case so you can cache the value and not call this every time the function is called.

Reply:
yeah that would make more sense, sorry for not understanding the result caching before